### PR TITLE
[Update] Create and edit geometries sample

### PR DESCRIPTION
**Pull request by des12437**
## Description

This PR updates `Create and edit geometries`
URL to README: https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/destiny/Update-Create-and-edit-geometries-sample/Shared/Samples/Create%20and%20edit%20geometries#readme

This PR updates the sample to match the sample design with the following changes:
- Updates the basemap and initial viewpoint
- Changes the styling of the graphics
- Adds initial graphics
- Allows deleting all graphics including the initial graphics

## Screenshots

create-and-edit-geometries-1  create-and-edit-geometries-2

### CODE REVIEW

**Review by dfeinzimer — APPROVED** [review]

File: Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift [hunk 1]
```diff
@@ -387,6 +415,115 @@ private class GeometryEditorModel: ObservableObject {
     }
 }
 
+private extension Geometry {
+    /// The area around the island of Inis Meáin (Aran Islands) in Ireland.
+    static var aranIslands: Envelope {
+        Envelope(center: Point(x: -9.5920, y: 53.08230, spatialReference: .wgs84), width: 1, height: 1, depth: 1)
+    }
+
```
Comment (yo1995) [inline Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift:423]:
```suggestion
```

File: Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift [hunk 2]
```diff
@@ -314,6 +321,28 @@ private class GeometryEditorModel: ObservableObject {
...
+        
+        let houseGraphic = Graphic(geometry: .house(), symbol: .point)
+        
+        geometryOverlay.addGraphics([
+            boundaryGraphic,
+            road1Graphic,
+            road2Graphic,
+            outbuildingsGraphic,
+            houseGraphic
+        ])
```
Comment (yo1995) [inline Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift:341]: The existing geometries are there for showing the editing workflow. Can you add the identify operation so that editing existing geometries also works, as described in the implementation details? Thank you. See around `onIdentifyGraphicsOverlayStatusChanged` symbol in the Qt example.

File: Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift [hunk 3]
```diff
@@ -413,24 +440,30 @@ private class GeometryEditorModel: ObservableObject {
         geometryEditor.start(withType: geometryType)
         isStarted = true
     }
+    
+    /// Starts editing a given graphic with the geometry editor.
+    /// - Parameter graphic: The graphic to edit.
+    func startEditing(withGraphic graphic: Graphic) {
+        guard let geometry = graphic.geometry else { return }
+        selectedGraphic = graphic
+        selectedGraphic?.isVisible = false
```
Comment (yo1995) [inline Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift:449]:
```suggestion
        graphic.isVisible = false
```

File: Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift [hunk 4]
```diff
@@ -364,6 +390,7 @@ private class GeometryEditorModel: ObservableObject {
     func stop() {
         geometryEditor.stop()
```
Comment (yo1995) [inline Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift:391]: After reading the implementation details, I like the idea to showcase that `stop` method returns the final geometry, because it might be something that a user won't discover by merely following this sample, without reading the doc.

Instead of hiding then removing the graphic being edited, can you modify the geometry instead? Thanks.

File: Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift [hunk 5]
```diff
@@ -380,11 +453,136 @@ private class GeometryEditorModel: ObservableObject {
     ///   - tool: The tool to draw with.
     ///   - geometryType: The type of geometry to draw.
     func startEditing(with tool: GeometryEditorTool, geometryType: Geometry.Type) {
+        selectedGraphic = nil
```
Comment (yo1995) [inline Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift:456]: Why is this line needed? 🤔

File: Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift [hunk 6]
```diff
@@ -380,11 +453,136 @@ private class GeometryEditorModel: ObservableObject {
...
+    /// - Parameter graphic: The graphic to edit.
+    func startEditing(withGraphic graphic: Graphic) {
+        guard let geometry = graphic.geometry else { return }
+        
+        switch geometry {
+        case is Point, is Multipoint:
+            geometryEditor.tool = VertexTool()
+        default:
+            break
+        }
```
Comment (yo1995) [inline Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift:473]: Is this needed?

File: Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift [hunk 7]
```diff
@@ -380,11 +453,136 @@ private class GeometryEditorModel: ObservableObject {
...
+        selectedGraphic = nil
         configureGeometryEditorTool(tool, scaleMode: scaleMode)
         geometryEditor.tool = tool
         geometryEditor.start(withType: geometryType)
         isStarted = true
     }
+    
+    /// Starts editing a given graphic with the geometry editor.
+    /// - Parameter graphic: The graphic to edit.
+    func startEditing(withGraphic graphic: Graphic) {
```
Comment (yo1995) [inline Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift:465]: I'm not entirely sure here, do you think it is Swift API Guideline compliant to name it `func startEditing(with graphic: Graphic)`?

File: Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift [hunk 8]
```diff
@@ -380,11 +453,136 @@ private class GeometryEditorModel: ObservableObject {
...
+        guard let geometry = graphic.geometry else { return }
```
Comment (yo1995) [inline Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift:466]:
```suggestion
        let geometry = graphic.geometry!
```
Since we can tap on the graphic, the geometry must be present.

**Review by yo1995 — COMMENTED** [review]
Overall looks good. A few minor questions

File: Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift (hunk 6 above)
Comment (des12437) [reply Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift:473]: Yes, it sets the geometry editor tool based on the geometry. This logic is specified in the sample design implementation details.

File: Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift (hunk 5 above)
Comment (des12437) [reply Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift:456]: This line can be removed.

File: Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift (hunk 7 above)
Comment (des12437) [reply Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift]: Removed the repeated type information from the method signature.

File: Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift (hunk 6 above)
Comment (yo1995) [reply Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift:472]: I'll ask the design author Andy why it is needed. Without these lines, the geometry editor still works, so I prefer removing them. Will let you know tomorrow if he gets back.

**Review by yo1995 — COMMENTED** [review]
LGTM with one thing to confirm.

**Review by dfeinzimer — DISMISSED** [review]

File: Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift (hunk 6 above)
Comment (des12437) [reply Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift]: Removed the unnecessary lines.

**Review by dfeinzimer — APPROVED** [review]

**Review by yo1995 — APPROVED** [review]

File: Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift (hunk 6 above)
Comment (yo1995) [reply Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift]: After consulting bistromath team, this logic needs to be added back.

**Comment by yo1995** [comment]
A update that I don't know where to put, so just leave here as a reference, in case we revisit these [changes](https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/452#discussion_r1683596409) in the future 🤔 

- If this logic is not added, there is a case such as using shape tool to edit `Point` will result in the GE not being able to handle the geometry so the geometry will be hidden by the GE
- If this logic is expanded to, say, edit every geometry with vertex tool, it totally works, but narrows down the tool to vertex tool will probably give a false indication to a user that one can only edit a geometry using vertex tool, which isn't true.
- Thus, the SDC decided although this logic is not perfect, it strikes a balance between being simple and being able to not surprise a user with unexpected behavior.
